extensions/cornerstone/src/init.js

```diff
@@ -275,3 +287,228 @@ const _connectToolsToMeasurementService = measurementService => {
+          const context = getNewContext(referenceElement.canvas);
+          context.setTransform(1, 0, 0, 1, 0, 0);
+          draw(context, newContext => {
+            drawLine(
+              newContext,
+              referenceElement.element,
+              referenceLine.start,
+              referenceLine.end,
+              { color: 'greenyellow' }
+            );
+          });
+
+          referenceElement.element.removeEventListener(
+            cornerstone.EVENTS.IMAGE_RENDERED,
+            onReferenceElementImageRendered
+          );
+        };
+
+        referenceElement.element.addEventListener(
+          cornerstone.EVENTS.IMAGE_RENDERED,
+          onReferenceElementImageRendered
+        );
+        cornerstone.updateImage(referenceElement.element);
+      });
+  };
+
+  cornerstone.events.addEventListener(
+    cornerstone.EVENTS.ELEMENT_DISABLED,
+    event => {
+      if (event.detail && event.detail.element) {
+        event.detail.element.removeEventListener(
+          cornerstone.EVENTS.IMAGE_RENDERED,
+          renderReferenceLines
+        );
+      }
+    }
+  );
+
+  const bindEnabledElementsEventListeners = enabledElements => {
+    enabledElements.forEach(enabledElement => {
+      enabledElement.addEventListener(
+        cornerstone.EVENTS.IMAGE_RENDERED,
+        renderReferenceLines
+      );
+    });
+  };
+
+  const unbindEnabledElementsEventListeners = enabledElements => {
+    enabledElements.forEach(enabledElement => {
+      enabledElement.removeEventListener(
+        cornerstone.EVENTS.IMAGE_RENDERED,
+        renderReferenceLines
+      );
+    });
+  };
+
+  let previousLayout;
+  window.store.subscribe(() => {
+    const { numColumns, numRows } = window.store.getState().viewports;
+    const viewportCount = numRows * numColumns;
+
+    if (viewportCount > 1) {
+      if (
+        !previousLayout ||
+        (previousLayout &&
+          (previousLayout.numColumns !== numColumns ||
+            previousLayout.numRows !== numRows))
+      ) {
+        previousLayout = { numColumns, numRows };
+
+        const enabledElements = [...csTools.store.state.enabledElements];
+
+        // We have all the elements
+        if (enabledElements.length === viewportCount) {
+          bindEnabledElementsEventListeners(enabledElements);
+        } else {
+          cornerstone.events.addEventListener(
+            cornerstone.EVENTS.ELEMENT_ENABLED,
+            ({ detail: { element } }) => {
+              enabledElements.push(element);
+
+              if (enabledElements.length === viewportCount) {
+                bindEnabledElementsEventListeners(enabledElements);
+              }
+            }
+          );
+        }
+      }
+    } else {
+      if (previousLayout !== undefined) {
+        previousLayout = undefined;
+        unbindEnabledElementsEventListeners(
+          csTools.store.state.enabledElements
+        );
+      }
+    }
+  });
+};
```

Comment:
Can we shift the bulk of this to a separate file to keep `init` clean?

Reply:
Done.